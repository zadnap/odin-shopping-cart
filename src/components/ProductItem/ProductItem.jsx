import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from './ProductItem.module.scss';
import Button from '@/components/Button/Button';
import { faStar, faTrash } from '@fortawesome/free-solid-svg-icons';
import { Link } from 'react-router-dom';

function ProductItem({
  id,
  title,
  rating,
  year,
  certification,
  backdropSrc,
  rent,
  rents,
  setRents,
}) {
  const handleCancelRenting = () => {
    const updatedRents = rents.filter((rentItem) => rentItem.id !== id);
    setRents(updatedRents);

    const storedIds = JSON.parse(localStorage.getItem('rents')) || [];
    const updatedIds = storedIds.filter((storedId) => storedId !== id);
    localStorage.setItem('rents', JSON.stringify(updatedIds));
  };

  return (
    <article className={styles.productItem}>
      <img
        className={styles.backdrop}
        src={backdropSrc}
        alt={`${title}'s backdrop`}
      />
      <div className={styles.info}>
        <div className={styles.header}>
          <Link to={`/movie/${id}`} className={styles.title} role="heading">
            {title}
          </Link>
          <div className={styles.movieMeta}>
            <p className={styles.tag}>{certification}</p> |
            <p className={styles.year}>{year}</p> |{' '}
            <p className={styles.rating}>
              <FontAwesomeIcon icon={faStar} /> <span>{rating}</span>
            </p>
          </div>
        </div>
        <div className={styles.buttonGroup}>
          <Button square onClick={handleCancelRenting}>
            <FontAwesomeIcon icon={faTrash} />
          </Button>
        </div>
        <p className={styles.rent}>${rent}</p>
      </div>
    </article>
  );
}

export default ProductItem;
