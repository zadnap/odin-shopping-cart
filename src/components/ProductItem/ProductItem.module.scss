@use '@/styles/variables.scss' as *;

.productItem {
  display: flex;
  gap: $gap;
  border-bottom: 1px solid $subbackground-color;
  padding: 15px 0;

  .backdrop {
    width: 10%;
    aspect-ratio: 1 / 1;
    object-fit: cover;
    border-radius: $border-radius;
  }

  .info {
    display: flex;
    flex-direction: column;
    justify-content: center;
    flex: 1;
    overflow: hidden;
    min-width: 0;
    gap: 3px;

    .title {
      font-size: 1.7rem;
      font-weight: 500;
      color: $text-color;
      display: block;
      min-width: 0;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;

      &:hover {
        text-decoration: underline;
      }
    }

    .movieMeta {
      display: flex;
      align-items: center;
      gap: 5px;
      font-size: 1.3rem;
      color: $subtext-color;

      .tag {
        font-size: 1.1rem;
        color: $text-color;
        border: 1px solid $subtext-color;
        border-radius: $border-radius;
        padding: 0 10px;
      }

      .year {
        color: $text-color;
      }

      .rating {
        color: yellow;
      }
    }
  }

  .actions {
    display: flex;
    align-items: center;
    gap: 10px;

    .removeButton {
      margin-left: auto;
      display: flex;
      justify-content: flex-end;
      gap: 10px;
    }

    .rent {
      font-size: 1.5rem;
    }
  }
}
