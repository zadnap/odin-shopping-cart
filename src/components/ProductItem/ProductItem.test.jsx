import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import ProductItem from './ProductItem';
import { MemoryRouter } from 'react-router-dom';

const mockProps = {
  id: 1,
  title: 'Lilo & Stitch',
  overview:
    'The wildly funny and touching story of a lonely Hawaiian girl and the fugitive alien who helps to mend her broken family.',
  rating: 7.1,
  year: 2025,
  certification: 'PG',
  posterSrc:
    'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/7c5VBuCbjZOk7lSfj9sMpmDIaKX.jpg',
  rent: 4.99,
};

describe('ProductItem component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <ProductItem {...mockProps} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
