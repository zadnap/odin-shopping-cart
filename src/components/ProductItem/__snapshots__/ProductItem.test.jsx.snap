// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductItem component > should render to match snapshot 1`] = `
<div>
  <article
    class="_productItem_b7926d"
  >
    <img
      alt="Lilo & Stitch's backdrop"
      class="_backdrop_b7926d"
    />
    <div
      class="_info_b7926d"
    >
      <div
        class="_header_b7926d"
      >
        <a
          class="_title_b7926d"
          data-discover="true"
          href="/movie/1"
          role="heading"
        >
          Lilo & Stitch
        </a>
        <div
          class="_movieMeta_b7926d"
        >
          <p
            class="_tag_b7926d"
          >
            PG
          </p>
           |
          <p
            class="_year_b7926d"
          >
            2025
          </p>
           |
           
          <p
            class="_rating_b7926d"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-star "
              data-icon="star"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                fill="currentColor"
              />
            </svg>
             
            <span>
              7.1
            </span>
          </p>
        </div>
      </div>
      <div
        class="_buttonGroup_b7926d"
      >
        <button
          class="
        _button_ee73f7 
         
        _square_ee73f7
        
      "
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash "
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7L128 32 32 32C14.3 32 0 46.3 0 64S14.3 96 32 96l384 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0-7.2-14.3C307.4 6.8 296.3 0 284.2 0L163.8 0c-12.1 0-23.2 6.8-28.6 17.7zM416 128L32 128 53.2 467c1.6 25.3 22.6 45 47.9 45l245.8 0c25.3 0 46.3-19.7 47.9-45L416 128z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <p
        class="_rent_b7926d"
      >
        $
        4.99
      </p>
    </div>
  </article>
</div>
`;
